ui/src/app/zones/views/ZoneDetails/ZoneDetailsHeader/ZoneDetailsHeader.tsx

```diff
@@ -0,0 +1,46 @@
+import { useEffect } from "react";
+
+import { Button } from "@canonical/react-components";
+import { useSelector, useDispatch } from "react-redux";
+
+import SectionHeader from "app/base/components/SectionHeader";
+import type { RootState } from "app/store/root/types";
+import { actions } from "app/store/zone";
+import zoneSelectors from "app/store/zone/selectors";
+
+type Props = {
+  id: number;
+};
+
+const ZoneDetailsHeader = ({ id }: Props): JSX.Element => {
+  const zonesLoaded = useSelector(zoneSelectors.loaded);
+  const zone = useSelector((state: RootState) =>
+    zoneSelectors.getById(state, Number(id))
+  );
+  const dispatch = useDispatch();
+
+  let buttons: JSX.Element[] | null = [
+    <Button appearance="neutral" data-test="delete-zone" key="delete-zone">
+      Delete AZ
+    </Button>,
+  ];
+
+  let title = "";
+
+  if (zonesLoaded && zone) {
+    title = `Availability zone: ${zone.name}`;
+  } else if (zonesLoaded) {
+    title = "Availability zone not found";
+    buttons = null;
+  }
+
+  useEffect(() => {
```

Comment:
My personal preference is to keep these one shot use-effects before the component logic as they are arguably the first thing that needs to execute.